pkg/controller/petset/pet_set_test.go

```diff
@@ -263,3 +263,12 @@ func TestPetSetBlockingPetIsCleared(t *testing.T) {
 		t.Errorf("Unexpected blocking pet, err %v: %+v", err, p)
 	}
 }
+
+func TestPetSetReplicaCount(t *testing.T) {
```

Comment:
I will add a test, I wanted this test to unit test only the low-level behaviour.